src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -3663,6 +3663,38 @@ instruct vsignum_reg(vReg dst, vReg zero, vReg one, vRegMask_V0 v0) %{
   ins_pipe(pipe_slow);
 %}
 
+// ---------------- Round float/double Vector Operations ----------------
+
+instruct vround_f(vReg dst, vReg src, fRegF tmp, vRegMask_V0 v0) %{
+  match(Set dst (RoundVF src));
+  effect(TEMP_DEF dst, TEMP tmp);
+  format %{ "java_round_float_v $dst, $src\t" %}
+  ins_encode %{
+    __ csrwi(CSR_FRM, C2_MacroAssembler::rdn);
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    __ vsetvli_helper(bt, Matcher::vector_length(this));
+    __ java_round_float_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg),
+                          as_FloatRegister($tmp$$reg));
+    __ csrwi(CSR_FRM, C2_MacroAssembler::rne);
```

Comment:
I don't think it's necessary to restore `CSR_FRM` to `rne` after `java_round_float_v`. As I remembered, we always set the required rounding mode in other places where it makes an effect. Similar for vround_d.

Reply:
Seems it should be unnecessary, but without it, 
1. there will test failure. e.g. test_suba which uses `vsub` intrinsic, and it does not set rounding mode explicitly.
2. an assert at `src/hotspot/os/linux/os_linux.cpp:1948` which is triggered at the end of program (when calling System.exit in the test).

Reply:
Ah, I see. I don't think it's safe for `vsub_fp`, `vdiv_fp`, etc to depend on some uncertain dynamic rounding mode on Java code entry. And it seems that the RISC-V ISA spec even doesn't specifies a default dynamic rounding mode. This reminds me that we might be lacking some pieces of the puzzle.

If we want to keep the `RNE` (Round to Nearest) rounding mode for Java code (like you do in this PR), we will also need to save and restore the default floating-point control state when we enter and leave Java code perfering `RNE`. Something similar for aarch64: https://bugs.openjdk.org/browse/JDK-8319973. Then we can remove the existing settings of rounding mode in file riscv_v.ad (grep "csrwi(CSR_FRM"), which should be good for performance. I would suggest we fix this in another PR before we continue with this one. Let me know if you are interested : -)

Reply:
Thanks for discussion.
Sure. Let me do some investigation and fix it first.

Reply:
tracked by https://bugs.openjdk.org/browse/JDK-8330094

Reply:
I have merged master (including https://github.com/openjdk/jdk/pull/18785, https://github.com/openjdk/jdk/pull/18758), and rerun the tests.

Reply:
Merged master again to include https://github.com/openjdk/jdk/pull/18839, and rerun the tests.